Accept parameters for creating a molecule from a notebook

bnmajor opened the pull request:
Users may optionally pass in parameters when importing a molecule in a notebook,
allowing them to set fields like 'name' and 'wikipediaUrl'.

![notebook_mol_params](https://user-images.githubusercontent.com/51238406/71996802-d2727680-320a-11ea-8ded-1e2e5590fda2.png)

bnmajor committed:
Accept parameters for creating a molecule from a notebook

Users may optionally pass in parameters when importing a molecule in a notebook,
allowing them to set fields like 'name' and 'wikipediaUrl'.

Signed-off-by: Brianna Major <brianna.major@kitware.com>

cjh1 commented on openchemistry/api.py:
```diff
@@ -88,13 +88,12 @@ def _calculation_monitor(taskflow_ids):
     return table
 
-def import_structure(smiles=None, inchi=None, cjson=None, gen3d=True):
+def import_structure(smiles=None, inchi=None, cjson=None, gen3d=True, params={}):
```
`params` has a mutable default value. The value will be assigned when the function is defined, so the same value will be used for each invocation that doesn't provide `params`, which is probably not intended as we are writing to it in the function. We should default it to `None` and then provide a new `dict` inside the function.

bnmajor replied:
Thank you for pointing this out! Fixed

bnmajor committed:
Do not use immutable value as default parameter

Using a mutable value as a default parameter not only mutates the value inside
the function, but also mutates the default parameter which will likely create
bugs. Use None as a default instead.

Signed-off-by: Brianna Major <brianna.major@kitware.com>

cryos reviewed (approved):
This looks good to me now that the None issue is resolved.
